<a href="https://colab.research.google.com/github/zeer0e1/estatistica-para-data-science-python/blob/main/Amostragem.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Amostragem estratificada

In [1]:
import pandas as pd

dataset = pd.read_csv('census.csv')

In [2]:
from sklearn.model_selection import StratifiedShuffleSplit

In [3]:
dataset['income'].value_counts()

 <=50K    24720
 >50K      7841
Name: income, dtype: int64

Executando o código acima temos a seguinte informação:

*   24720 pessoas ganham menos de 50 mil
*   7841 pessoas ganham mais de 50 mil

se dividirmos a quantidade de pessoas que ganham mais de 50 mil pelo tamanho da população vamos obter a porcentagem que essas pessoas representam na população

In [6]:

7841 / len(dataset), 24720 / len(dataset)

(0.2408095574460244, 0.7591904425539756)

Executanndo o código acima temos a informação que

*   24% da população recebe um salário maior que 50 mil
*   75% da população recebe um salário menor que 50 mil

Por isso temos que para retirar uma amostra temos que utilizar uma amostra estratificada pois a probabilidade de dentro amostr possuir somente pessoas que recebem menos que 50 mil é muito grando pois eles estão em maior número na base